packages/hdwallet-ledger/src/bitcoin.ts

```diff
@@ -173,18 +173,23 @@ export async function btcSignTx(
     if (msg.coin === "BitcoinCash" && bchAddr.isCashAddress(outputAddress)) {
       outputAddress = bchAddr.toLegacyAddress(outputAddress);
     }
-    txBuilder.addOutput(outputAddress, Number(output.amount));
+    psbt.addOutput({ address: outputAddress, value: BigInt(output.amount) });
   }
 
   if (msg.opReturnData) {
     if (msg.opReturnData.length > 80) {
       throw new Error("OP_RETURN data must be less than 80 chars.");
     }
-    const ret = bitcoin.script.compile([bitcoin.opcodes.OP_RETURN, Buffer.from(msg.opReturnData)]);
-    txBuilder.addOutput(ret, 0);
+    const script = bitcoin.script.compile([bitcoin.opcodes.OP_RETURN, Buffer.from(msg.opReturnData)]);
+
+    psbt.addOutput({
+      script,
+      // OP_RETURN_DATA outputs always have a value of 0
+      value: BigInt(0),
+    });
```

Comment:
The same as before but using Psbt class expected shape, including value as BigInt